### PR TITLE
feat(webpack-dev-server): Improve dev experience with webpack dev server

### PR DESCRIPTION
**Pull request by mattsjones**
- Add live reload
- Add error overlay 
- Show only error logs 
- Support custom hosts 
- Make port configurable 
- Add historyApiFallback

### CODE REVIEW

File: scripts/start.js [hunk 1]
```diff
@@ -2,22 +2,21 @@ const WebpackDevServer = require('webpack-dev-server');
...
+const args = require('../config/args');
 
 const compiler = webpack(webpackConfig);
 const devServer = new WebpackDevServer(compiler, {
-  contentBase: builds.map(({ paths }) => paths.public)
+  contentBase: builds.map(({ paths }) => paths.public),
+  historyApiFallback: true,
+  overlay: true,
+  stats: 'errors-only',
+  allowedHosts: ['dev.seek.com.au', 'dev.seek.co.au']
```
Comment (markdalgleish) [inline scripts/start.js]: Typo, should be `nz` on the last one.

File: config/webpack/webpack.config.js [hunk 2]
```diff
@@ -121,9 +121,20 @@ const buildWebpackConfigs = builds.map(
 
     const internalJs = [paths.src, ...paths.compilePackages];
 
+    const entry = [paths.clientEntry];
+    const devEntries = [
+      `${require.resolve(
+        'webpack-dev-server/client'
+      )}?http://localhost:${port}/`
```
Comment (markdalgleish) [inline config/webpack/webpack.config.js]: Will this localhost reference work if localhost isn't in the list of allowed hosts?

Comment (mattsjones) [inline config/webpack/webpack.config.js]: Yes

Comment (markdalgleish) [inline config/webpack/webpack.config.js]: 🆗 🆒 👍

File: README.md [hunk 3]
```diff
@@ -295,6 +295,21 @@ will create `index-AU.html` & `index-NZ.html`.
 
 Note: When running the app in dev mode only one HTML file will be created, defaulting to the first listed locale.
 
+### Development server
+
+Out of the box sku will start your app with [webpack-dev-server]() on http://localhost:8080. However there a few options you can pass `sku.config.js` if needed.
```
Comment (markdalgleish) [inline README.md]: Missing URL for the webpack-dev-server link?

Comment (mattsjones) [inline README.md]: I wrote this thinking *I need to remember to fill this in* :facepalm:

**Pull request merged**